### ISSUE TITLE
pylint should not use Class name regex for TypeVar names

### ISSUE THREAD

**Issue by jnsnow** [PyCQA/pylint#3401]
### Steps to reproduce

1. Consider the following example:

```python
"""Example"""
from typing import TypeVar

T = TypeVar('T')


def identity(arg: T) -> T:
    """Identity function."""
    return arg
```

2. pylint typevar.py

### Current behavior

typevar.py:4:0: C0103: Class name "T" doesn't conform to PascalCase naming style (invalid-name)

### Expected behavior

Either:
- Test should pass
- pylint should use a whitelist of names for TypeVars that is separate from that which is used for Class names. 'T' is a perfectly canonical TypeVar annotation in functional notation, but it is a perfectly rubbish runtime Class name.

See https://docs.python.org/3/library/typing.html#typing.TypeVar for a canonical example of usage from the official python docs.

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.7.6 (default, Dec 19 2019, 22:50:01) 
[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)]
```

**Comment by remort** [comment]
Still hurts.
Can we workaround this bug in .pylintrc or somehow by far ?
Our broad code base suffers from this.

**Comment by hippo91** [comment]
@remort there is the PR #3401 that should fix this issue. However i'v got no answer from the author (@kaihoankalina).
In the meantime maybe you can have a look to the `class-rgx` parameter inside your `.pylintrc` file.

**Comment by remort** [comment]
As I get it - changing `class-rgx` default value leads to changes in linting logic for both: Classes and module level Constants at the same time? If so - this is an undesirable solution.
